### PR TITLE
Updated documentation of arangoimport

### PR DESCRIPTION
**Pull request by cpjulia**
This PR adds more details to arangoimport documentation, showing the link of the naming convention for `_key` and a warning of race condition of parallelism and impredictability of the documents that will be inserted when there are duplicates.
The trigger for making this PR was a case of our internal usage of arangoimport to read a csv file with this command line (some details suppressed on purpose)
```
./build/bin/arangoimport  --file ./<fileName>.csv --type csv --collection "routers" --create-collection true --skip-validation true --overwrite true --translate hostname=_key --log.level warning --ignore-missing true --on-duplicate "ignore"
```
the field `hostname` that was mapped to `_key` had many duplicate values, and, even with setting `--on-duplicate "ignore"`, there would still be warnings about the duplicates and unpredictability of what document would be inserted (as only one document with the same `_key` could be inserted because of the uniqueness constraint). This was due to a race condition in multithreading, so it should be warned in the documentation that, for such case scenarios, the command should be run with a single thread, e.g., `--threads 1`. 
Also, in the csv file, the values for the field that would be mapped to `_key` had non permitted symbols, like `Ã`. 

So this scenario suggested an update in the documentation to warn about best practices when using arangoimport in the command line.

### CODE REVIEW

File: 3.10/programs-arangoimport-details.md [hunk 1]
```diff
@@ -174,7 +175,10 @@ Automatic pacing intentionally may not use the full throughput of a
 disk device. An unlimited (really fast) disk device might not need
 pacing. Raising the number of threads via the `--threads X` command
 line to any value of `X` greater than 2 will increase the total
-throughput used.
+throughput used. <strong>Warning</strong>: using parelellism with the
+parameter `--threads X` can lead to a race condition when there are duplicates
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the
+output unpredictable, meaning not knowing which document will be inserted.
```
Comment (jsteemann) [inline 3.10/programs-arangoimport-details.md:181]:
```suggestion
throughput used. 

<strong>Warning</strong>: using parellelism with the parameter `--threads X` 
together with the parameter `--on-duplicate` set to `ignore`, `update` or `replace` can 
lead to a race condition when there are duplicates e.g. multiple identical `_key`
values. Even ignoring the duplicates will make the result unpredictable, meaning 
it is not predictable which versions of the documents will be inserted.
```

File: 3.10/programs-arangoimport-examples-json.md [hunk 2]
```diff
@@ -245,4 +245,7 @@ Using multiple threads may lead to a non-sequential import of the input
 data. Data that appears later in the input file may be imported earlier than data
 that appears earlier in the input file. This is normally not a problem but may cause
 issues when when there are data dependencies or duplicates in the import data. In
-this case, the number of threads should be set to 1.
+this case, the number of threads should be set to 1. Also, using parallelism with the
+parameter `--threads X` can lead to a race condition when there are duplicates
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the
+output unpredictable, meaning not knowing which document will be inserted.
```
Comment (jsteemann) [inline 3.10/programs-arangoimport-examples-json.md:251]:
```suggestion
parameter `--threads X` together with the parameter `--on-duplicate` set to `ignore`, 
`update` or `replace` can lead to a race condition when there are duplicates e.g. 
multiple identical `_key` values. Even ignoring the duplicates will make the result 
unpredictable, meaning it is not predictable which versions of the documents will be inserted.

File: 3.7/programs-arangoimport-details.md [hunk 3]
```diff
@@ -175,7 +176,10 @@ Automatic pacing intentionally may not use the full throughput of a
 disk device. An unlimited (really fast) disk device might not need
 pacing. Raising the number of threads via the `--threads X` command
 line to any value of `X` greater than 2 will increase the total
-throughput used.
+throughput used. <strong>Warning</strong>: using parelellism with the
+parameter `--threads X` can lead to a race condition when there are duplicates
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the
+output unpredictable, meaning not knowing which document will be inserted.
```
Comment (jsteemann) [inline 3.7/programs-arangoimport-details.md:182]:
```suggestion
throughput used. 

<strong>Warning</strong>: using parellelism with the parameter `--threads X` 
together with the parameter `--on-duplicate` set to `ignore`, `update` or `replace` can 
lead to a race condition when there are duplicates e.g. multiple identical `_key`
values. Even ignoring the duplicates will make the result unpredictable, meaning 
it is not predictable which versions of the documents will be inserted.
```

File: 3.7/programs-arangoimport-examples-json.md [hunk 4]
```diff
@@ -246,4 +246,7 @@ Using multiple threads may lead to a non-sequential import of the input
 data. Data that appears later in the input file may be imported earlier than data
 that appears earlier in the input file. This is normally not a problem but may cause
 issues when when there are data dependencies or duplicates in the import data. In
-this case, the number of threads should be set to 1.
+this case, the number of threads should be set to 1. Also, using parallelism with the
+parameter `--threads X` can lead to a race condition when there are duplicates
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the
+output unpredictable, meaning not knowing which document will be inserted.
```
Comment (jsteemann) [inline 3.7/programs-arangoimport-examples-json.md:252]:
```suggestion
parameter `--threads X` together with the parameter `--on-duplicate` set to `ignore`, 
`update` or `replace` can lead to a race condition when there are duplicates e.g. 
multiple identical `_key`values. Even ignoring the duplicates will make the result 
unpredictable, meaning it is not predictable which versions of the documents will be inserted.
```

File: 3.8/programs-arangoimport-details.md [hunk 5]
```diff
@@ -174,7 +175,10 @@ Automatic pacing intentionally may not use the full throughput of a
 disk device. An unlimited (really fast) disk device might not need
 pacing. Raising the number of threads via the `--threads X` command
 line to any value of `X` greater than 2 will increase the total
-throughput used.
+throughput used. <strong>Warning</strong>: using parelellism with the
+parameter `--threads X` can lead to a race condition when there are duplicates
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the
+output unpredictable, meaning not knowing which document will be inserted.
```
Comment (jsteemann) [inline 3.8/programs-arangoimport-details.md:181]:
```suggestion
throughput used.

<strong>Warning</strong>: using parellelism with the parameter `--threads X` 
together with the parameter `--on-duplicate` set to `ignore`, `update` or `replace` can 
lead to a race condition when there are duplicates e.g. multiple identical `_key`
values. Even ignoring the duplicates will make the result unpredictable, meaning 
it is not predictable which versions of the documents will be inserted.
```

File: 3.8/programs-arangoimport-examples-json.md [hunk 6]
```diff
@@ -245,4 +245,7 @@ Using multiple threads may lead to a non-sequential import of the input
 data. Data that appears later in the input file may be imported earlier than data
 that appears earlier in the input file. This is normally not a problem but may cause
 issues when when there are data dependencies or duplicates in the import data. In
-this case, the number of threads should be set to 1.
+this case, the number of threads should be set to 1. Also, using parallelism with the
+parameter `--threads X` can lead to a race condition when there are duplicates
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the
+output unpredictable, meaning not knowing which document will be inserted.
```
Comment (jsteemann) [inline 3.8/programs-arangoimport-examples-json.md:251]:
```suggestion
parameter `--threads X` together with the parameter `--on-duplicate` set to `ignore`, 
`update` or `replace` can lead to a race condition when there are duplicates e.g. 
multiple identical `_key`values. Even ignoring the duplicates will make the result 
unpredictable, meaning it is not predictable which versions of the documents will be inserted.
```

File: 3.9/programs-arangoimport-details.md [hunk 7]
```diff
@@ -174,7 +175,10 @@ Automatic pacing intentionally may not use the full throughput of a
 disk device. An unlimited (really fast) disk device might not need
 pacing. Raising the number of threads via the `--threads X` command
 line to any value of `X` greater than 2 will increase the total
-throughput used.
+throughput used. <strong>Warning</strong>: using parelellism with the
+parameter `--threads X` can lead to a race condition when there are duplicates 
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the 
+output unpredictable, meaning not knowing which document will be inserted.
```
Comment (jsteemann) [inline 3.9/programs-arangoimport-details.md:181]:
```suggestion
throughput used. 

<strong>Warning</strong>: using parellelism with the parameter `--threads X` 
together with the parameter `--on-duplicate` set to `ignore`, `update` or `replace` can 
lead to a race condition when there are duplicates e.g. multiple identical `_key`
values. Even ignoring the duplicates will make the result unpredictable, meaning 
it is not predictable which versions of the documents will be inserted.
```

File: 3.9/programs-arangoimport-examples-json.md [hunk 8]
```diff
@@ -245,4 +245,9 @@ Using multiple threads may lead to a non-sequential import of the input
 data. Data that appears later in the input file may be imported earlier than data
 that appears earlier in the input file. This is normally not a problem but may cause
 issues when when there are data dependencies or duplicates in the import data. In
-this case, the number of threads should be set to 1.
+this case, the number of threads should be set to 1. Also, using parallelism with the
+parameter `--threads X` can lead to a race condition when there are duplicates 
+e.g. multiple equal `_key`values, so even ignoring the duplicates will make the 
+output unpredictable, meaning not knowing which document will be inserted.  
+
+
```
Comment (jsteemann) [inline 3.9/programs-arangoimport-examples-json.md:253]:
```suggestion
this case, the number of threads should be set to 1. Also, using parallelism with the
parameter `--threads X` together with the parameter `--on-duplicate` set to `ignore`, 
`update` or `replace` can lead to a race condition when there are duplicates e.g. 
multiple identical `_key`values. Even ignoring the duplicates will make the result 
unpredictable, meaning it is not predictable which versions of the documents will be inserted.
```

**Review by jsteemann — APPROVED** [review]
LGTM!

**Review by ansoboleva — APPROVED** [review]
reviewed

**Pull request merged**